javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/SitewiseActions.java

```diff
@@ -0,0 +1,479 @@
+        DeleteAssetModelRequest deleteAssetModelRequest = DeleteAssetModelRequest.builder()
+            .assetModelId(assetModelId)
+            .build();
+
+        return getAsyncClient().deleteAssetModel(deleteAssetModelRequest)
+            .whenComplete((response, exception) -> {
+                if (exception != null) {
+                    throw new RuntimeException("Failed to delete asset model with ID: " + assetModelId + ". Error: " + exception.getMessage(), exception);
+                }
+            });
+    }
+    // snippet-end:[sitewise.java2.delete.asset.model.main]
+
+    // snippet-start:[sitewise.java2.create.portal.main]
+    /**
+     * Creates a new IoT SiteWise portal asynchronously.
+     *
+     * @param portalName  the name of the portal to create
+     * @param iamRole     the IAM role ARN to use for the portal
+     * @param contactEmail the email address of the portal contact
+     * @return a {@link CompletableFuture} that completes with the portal ID when the portal is created successfully, or throws a {@link RuntimeException} if the creation fails
+     */
+    public CompletableFuture<String> createPortalAsync(String portalName, String iamRole, String contactEmail) {
+        CreatePortalRequest createPortalRequest = CreatePortalRequest.builder()
+            .portalName(portalName)
+            .portalDescription("This is my custom IoT SiteWise portal.")
+            .portalContactEmail(contactEmail)
+            .roleArn(iamRole)
+            .build();
+
+        return getAsyncClient().createPortal(createPortalRequest)
+            .handle ((response, exception) -> {
+                if (exception != null) {
+                    logger.error("Failed to create portal: {} ", exception.getCause().getMessage());
+                    throw (CompletionException) exception;
+                }
+                return response.portalId();
+            });
+    }
+    // snippet-end:[sitewise.java2.create.portal.main]
+
+    // snippet-start:[sitewise.java2.delete.portal.main]
+    /**
+     * Deletes a portal asynchronously.
+     *
+     * @param portalId the ID of the portal to be deleted
+     * @return a {@link CompletableFuture} containing the {@link DeletePortalResponse} when the operation is complete
+     * @throws RuntimeException if the portal deletion fails, with the error message and the underlying exception
+     */
+    public CompletableFuture<DeletePortalResponse> deletePortalAsync(String portalId) {
+        DeletePortalRequest deletePortalRequest = DeletePortalRequest.builder()
+            .portalId(portalId)
+            .build();
+
+        return getAsyncClient().deletePortal(deletePortalRequest)
+            .whenComplete((response, exception) -> {
+                if (exception != null) {
+                    throw new RuntimeException("Failed to delete portal with ID: " + portalId + ". Error: " + exception.getMessage(), exception);
+                }
```

Comment:
just log, not rethrow.

Reply:
The code team wants exceptions bubbled up to calling code